File: typescript/src/model/expressions/wdl-array-literal.ts
/** WDL array literal expression nodes. */
import { WdlExpressionComponentType, type WdlExpression } from './wdl-expression.js';

/** Models an ordered array literal such as `[1, 2, 3]`. */
export class WdlArrayLiteral implements WdlExpression {
  private readonly entryValues: WdlExpression[] = [];

  /** Returns the ordered element expressions contained in the literal. */
  public entries(): WdlExpression[] {
    return this.entryValues;
  }

  /** Returns the broad expression family for this node. */
  public componentType(): WdlExpressionComponentType {
    return WdlExpressionComponentType.ARRAY_LIT;
  }
}

File: java/src/main/java/com/myriad/wdl/model/expressions/WdlArrayLiteral.java
package com.myriad.wdl.model.expressions;

import java.util.ArrayDeque;

public final class WdlArrayLiteral implements WdlExpression {
  private final ArrayDeque<WdlExpression> entries = new ArrayDeque<>();

  public ArrayDeque<WdlExpression> entries() {
    return entries;
  }

  @Override
  public String toString() {
    return getClass().getSimpleName();
  }

  @Override
  public ComponentType componentType() {
    return ComponentType.ARRAY_LIT;
  }
}
